packages/theming-core/src/generics/themeRegistry.test.ts

```diff
@@ -0,0 +1,125 @@
+import { createThemeRegistry, IThemeRegistry } from './themeRegistry';
+
+interface IPartialTestTheme {
+  val: number;
+}
+
+interface ITestTheme {
+  val: number;
+  sum: number;
+}
+
+function _resolveTestTheme(definition: IPartialTestTheme | undefined, parent: ITestTheme): ITestTheme {
+  const val: number = (definition && definition.val) || 0;
+  return {
+    val,
+    sum: parent.sum + val
+  };
+}
+
+function _addTen(theme: ITestTheme): IPartialTestTheme {
+  return {
+    val: theme.val + 10
+  };
+}
+
+function _testRegistry(): IThemeRegistry<ITestTheme, IPartialTestTheme> {
+  return createThemeRegistry<ITestTheme, IPartialTestTheme>({ val: 0, sum: 0 }, _resolveTestTheme);
+}
+
+describe('theme registry tests', () => {
```

Comment:
Convention on tests note:

`describe` is used to describe a Thing. `it` is used to describe what the Thing does. Logging will likely concatenate these together. So consider:

```ts
describe('themeRegistry', () => {
  it('can do X', () => { ... });
  it('should call N when Z happens', () => { ... });
});
```
might read i logging (or error messages):
```
themeRegistry can do X
themeRegistry should call N when Z happens
```